meerkat_frontend/src/js/technical/tableManager.js

```diff
@@ -659,6 +659,182 @@ function addPaginationListener(table) {
     });
 }
 
+/**:drawContactSummaryTable(containerID, location_id)
+
+    :param string containerID:
+        The ID attribute of the html element to hold the table.
+    :param number location_id:
+        The ID of the location by which to filer data.
+ */
+function drawContactSummaryTable(containerID, location_id) {
+    console.log("Now, I'm here");
+    $.getJSON(api_root + "/locations", function(locations) {
+        $.getJSON(api_root + "/aggregate_category_sum/cv_visits_reason/" + location_id, function(con_reasons) {
+            $.getJSON(api_root + "/aggregate_category_sum/cv_visits/" + location_id, function(con_visits) {
+                $.getJSON(api_root + "/aggregate_category/key_indicators/" + location_id, function(totals) {
+                    $.getJSON(api_root + "/variables/cv_visits_reason", function(con_reasons_names) {
```

Comment:
There's quite a lot of JSON requests here.  You could parallelise them like in the category summation function.  I can talk you through this if you want, it is fairly simple and will speed the page load up significantly.